augur/filters/exclude_by_nonnucleotide_symbols.py

```diff
@@ -0,0 +1,53 @@
+import functools
+
+from augur.filters.base_filter import BaseFilter
+
+
+def is_enabled(args):
+    return args.non_nucleotide
+
+
+def add_arguments(parser):
+    parser.add_argument(
+        "--non-nucleotide",
+        action="store_true",
+        help="exclude sequences that contain illegal characters",
+    )
+
+
+def init(context):
+    return ExcludeByNonnucleotideSymbols()
+
+
+class ExcludeByNonnucleotideSymbols(BaseFilter):
+    GOOD_SYMBOLS = {
+        "A",
+        "C",
+        "G",
+        "T",
+        "-",
+        "N",
+        "R",
+        "Y",
+        "S",
+        "W",
+        "K",
+        "M",
+        "D",
+        "H",
+        "B",
+        "V",
+        "?",
+    }
+
+    def affected_sequences(self, sequences):
+        # TODO probably a potential speedup here
+        return [
+            s
+            for s in sequences
+            if len(set(str(s.sequence).upper()).difference(self.GOOD_SYMBOLS)) != 0
+        ]
```

Comment:
I've got similar "good nucleotide" code in #459 - think the faster route, since we Just need to check if any nucleotide is bad, is:
```python
GOOD_SYMBOLS = {"a","t",etc}
return any(site not in GOOD_SYMBOLS for site in s.sequence)
```
effectively, this will stop immediately upon finding a bad nucleotide, instead of parsing the rest of the sequence